libtransact/src/workload/xo.rs

```diff
@@ -0,0 +1,182 @@
+use crate::transaction::TransactionPair;
+use crate::workload::error::WorkloadError;
+use crate::workload::BatchWorkload;
+use crate::workload::TransactionWorkload;
+
+use rand::distributions::Alphanumeric;
+use rand::prelude::*;
+use rand_hc::Hc128Rng;
+
+use sha2::{Digest, Sha512};
+
+static FAMILY_NAME: &str = "xo";
+static FAMILY_VERSION: &str = "0.1";
+static NONCE_SIZE: usize = 32;
+
+pub struct XoTransactionWorkload {
+    rng: Hc128Rng,
+    signer: Box<Signer>,
+}
+
+impl XoTransactionWorkload {
+    pub fn new() -> Self {
+        let rng = Hc128Rng::from_entropy();
+        let signer = HashSigner::new();
+
+        XoTransactionWorkload {
+            rng,
+            signer: Box::new(signer),
+        }
+    }
+
+    pub fn new_with_seed(seed: u64) -> XoTransactionWorkload {
+        let rng = Hc128Rng::seed_from_u64(seed);
+        let signer = HashSigner::new();
+
+        XoTransactionWorkload {
+            rng,
+            signer: Box::new(signer),
+        }
+    }
+}
+
+impl TransactionWorkload for XoTransactionWorkload {
+    fn next_transaction(&mut self) -> Result<TransactionPair, WorkloadError> {
+        let nonce = self
+            .rng
+            .sample_iter(&Alphanumeric)
+            .take(NONCE_SIZE)
+            .collect::<String>()
+            .as_bytes()
+            .to_vec();
+
+        let payload = Payload::new_as_create_with_random_name(&mut self.rng);
+
+        Ok(TransactionBuilder::new()
+            .with_family_name(FAMILY_NAME.to_string())
+            .with_family_version(FAMILY_VERSION.to_string())
+            .with_inputs(payload.inputs())
+            .with_outputs(payload.outputs())
+            .with_nonce(nonce)
+            .with_payload(payload.bytes())
+            .with_payload_hash_method(HashMethod::SHA512)
+            .build_pair(&*self.signer)?)
+    }
+}
+
+pub struct XoBatchWorkload {
+    transaction_workload: XoTransactionWorkload,
+    signer: Box<Signer>,
+}
+
+impl XoBatchWorkload {
+    pub fn new() -> XoBatchWorkload {
+        let signer = HashSigner::new();
+
+        XoBatchWorkload {
+            transaction_workload: XoTransactionWorkload::new(),
+            signer: Box::new(signer),
+        }
+    }
+
+    pub fn new_with_seed(seed: u64) -> XoBatchWorkload {
+        let signer = HashSigner::new();
+
+        XoBatchWorkload {
+            transaction_workload: XoTransactionWorkload::new_with_seed(seed),
+            signer: Box::new(signer),
+        }
+    }
+}
+
+impl BatchWorkload for XoBatchWorkload {
+    fn next_batch(&mut self) -> Result<BatchPair, WorkloadError> {
+        Ok(BatchBuilder::new()
+            .with_transactions(vec![self.transaction_workload.next_transaction()?.take().0])
+            .build_pair(&*self.signer)?)
+    }
+}
+
+enum Action {
+    CREATE,
+}
+
+struct Payload {
+    name: String,
+    action: Action,
+}
+
+impl Payload {
+    pub fn new_as_create_with_random_name(rnd: &mut Hc128Rng) -> Self {
```

Comment:
Can you add some doc comments to these `new` functions? It is not entirely obvious what `as_create` means.

Reply:
Added